lib/matplotlib/axes/_axes.py

```diff
@@ -6409,23 +6409,32 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
              orientation='vertical', rwidth=None, log=False,
              color=None, label=None, stacked=False, **kwargs):
         """
-        Plot a histogram.
+        Compute and plot a histogram.
 
-        Compute and draw the histogram of *x*.  The return value is a tuple
-        (*n*, *bins*, *patches*) or ([*n0*, *n1*, ...], *bins*, [*patches0*,
-        *patches1*, ...]) if the input contains multiple data.  See the
-        documentation of the *weights* parameter to draw a histogram of
-        already-binned data.
+        This method uses `numpy.histogram` to bin the data in *x* and count the
+        number of values in each bin, then draws the distribution either as a
+        `.BarContainer` or `.Polygon`. The *bins*, *range*, *density*, and
+        *weights* parameters are forwarded to `numpy.histogram`. If the data
+        has already been binned and counted, we recommend using `~.bar` or
```

Comment:
Minor editorial point: I prefer keeping an impersonal style (who is "we"? am I responsible for each of the recommendations made in the name of "we"?), e.g. "If the data has already been binned and counted, consider using bar or stairs..."

Reply:
Personal opinion: "we" - the Matplotlib development team. It's a collective responsibility, not an individual personal one. Same as for code, API or existing examples. I think there is no semantic difference between a personal and impersonal style for recommendations. To me the impersonal passive sounds a bit more convoluted.

Currently we use the impersonal style more often than "we". By me, this is ok to change to impersonal here. Possibly we should have a discussion about this at some point and add the result to https://matplotlib.org/stable/devel/style_guide.html, but low priority.

Reply:
The other option, by the way, would be to just drop the "we recommend" and go straight to the point: "if .. already binned, use bar or stairs..."